### PR TITLE
Update `genref` version

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Update `genref` version.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by trasc** [comment]
/hold 
For a genref release tag.

https://github.com/kubernetes-sigs/reference-docs/pull/343#issuecomment-1784550238

File: Makefile [hunk 1]
```diff
@@ -321,8 +321,8 @@ helm: ## Download helm locally if necessary.
 GENREF = $(PROJECT_DIR)/bin/genref
 .PHONY: genref
 genref: ## Download genref locally if necessary.
-	# TODO: update the package version. "c519be1" - is the master tip on 27-Oct-2023
-	@GOBIN=$(PROJECT_DIR)/bin $(GO_CMD) install github.com/kubernetes-sigs/reference-docs/genref@c519be1
+	# TODO: update the package version. "c519be1" - is the master tip on 30-Oct-2023
+	@GOBIN=$(PROJECT_DIR)/bin $(GO_CMD) install github.com/kubernetes-sigs/reference-docs/genref@23a78f4
```
Comment (tenzen-y) [inline Makefile:325]:
```suggestion
	@GOBIN=$(PROJECT_DIR)/bin $(GO_CMD) install github.com/kubernetes-sigs/reference-docs/genref@v0.28.0
```

Can we use the released version?

https://github.com/kubernetes-sigs/reference-docs/releases/tag/v0.28.0

This version seems to have your patch.

**Comment by tenzen-y** [comment]
> /hold For a genref release tag.
> 
> [kubernetes-sigs/reference-docs#343 (comment)](https://github.com/kubernetes-sigs/reference-docs/pull/343#issuecomment-1784550238)

Oh, we found the released version at the same time :)

**Pull request merged**